android-json-form-wizard/src/main/java/com/vijay/jsonwizard/filesource/AssetsFileSource.java

```diff
@@ -0,0 +1,67 @@
+package com.vijay.jsonwizard.filesource;
+
+import android.content.Context;
+
+import com.vijay.jsonwizard.interfaces.FormFileSource;
+
+import org.jeasy.rules.api.Rules;
+import org.jeasy.rules.mvel.MVELRuleFactory;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import timber.log.Timber;
+
+/***
+ * used to read files stored on the APP's asset folder
+ *
+ */
+public class AssetsFileSource implements FormFileSource {
+
+    public static AssetsFileSource INSTANCE = new AssetsFileSource();
+
+    private AssetsFileSource() {
+    }
+
+    @Override
+    public Rules getRulesFromFile(Context context, String fileName) throws Exception {
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(context.getAssets().open(fileName)));
+        return MVELRuleFactory.createRulesFrom(bufferedReader);
+    }
+
+    @Override
+    public JSONObject getFormFromFile(Context context, String fileName) {
+        try {
+            String newFileName = "json.form/" + fileName + ".json";
+
+            InputStream inputStream = context.getAssets()
+                    .open(newFileName);
+
+            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream,
+                    "UTF-8"));
+            String jsonString;
+            StringBuilder stringBuilder = new StringBuilder();
+            while ((jsonString = reader.readLine()) != null) {
```

Comment:
Why not use `convertStreamToString` here?

Reply:
Wouldn't it be easier to move the file reader utils to the native form library then?

This will be useful by other apps as well.